packages/@uppy/core/src/Uppy.js

```diff
@@ -1242,7 +1080,11 @@ class Uppy {
 
     this.on('dashboard:file-edit-complete', (file) => {
       if (file) {
-        this.#checkRequiredMetaFieldsOnFile(file)
+        const ctx = {
+          setFileState: (...args) => this.setFileState(...args),
```

Comment:
Alternatively, we could define a private property that to make it clear it's a bound method
```suggestion
          setFileState: this.#setFileStateBound,
```

And add below the `setFileState` definition:

```js
  #setFileStateBound = (...args) => this.setFileState(...args)
```